src/extension.ts

```diff
@@ -172,7 +173,9 @@ export function activate(context: vscode.ExtensionContext): void {
         }
 
         const createdApp = <IAzureNode<WebAppTreeItem>>await node.createChild(this);
-
+        const createdAppConfig: SiteConfigResource = await createdApp.treeItem.client.getSiteConfig();
+        this.properties.os = createdAppConfig.kind ? createdAppConfig.kind : 'undefined';
```

Comment:
Not sure why you're not just leaving it undefined (which will result in a null entry in telemetry).

Reply:
Does it automatically convert the string "undefined" to a null entry?  Definitely didn't want that.

Reply:
Perhaps we should change the type of `TelemetryProperties` [here](https://github.com/Microsoft/vscode-azuretools/blob/master/ui/index.d.ts#L240) to `string | undefined` instead of just `string`.

That way you could do `this.properties.os = createdAppConfig.kind` without ts yelling at you